src/vs/workbench/contrib/terminal/common/scripts/shellIntegration-bash.sh

```diff
@@ -231,37 +231,35 @@ __vsc_update_cwd() {
 	builtin printf '\e]633;P;Cwd=%s\a' "$(__vsc_escape_value "$__vsc_cwd")"
 }
 
+# Filtered env output without exported functions
+__vsc_env() {
+	command env -0 | tr '\0\n' '\n\r' | sort | grep -Ev '^BASH_FUNC'
+}
+
 __updateEnvCacheAA() {
-	local key="$1"
-	local value="$2"
-	if [ "$use_associative_array" = 1 ]; then
-		if [[ "${vsc_aa_env[$key]}" != "$value" ]]; then
-			vsc_aa_env["$key"]="$value"
-			builtin printf '\e]633;EnvSingleEntry;%s;%s;%s\a' "$key" "$(__vsc_escape_value "$value")" "$__vsc_nonce"
-		fi
+	local key="$1" value="$2"
+	if [[ "${vsc_aa_env[$key]-}" != "$value" ]]; then
+		vsc_aa_env["$key"]="$value"
+		builtin printf '\e]633;EnvSingleEntry;%s;%s;%s\a' "$key" "$(__vsc_escape_value "$value")" "$__vsc_nonce"
 	fi
 }
 
 __trackMissingEnvVarsAA() {
-	if [ "$use_associative_array" = 1 ]; then
```

Comment:
I think we still need the check here for older versions of bash? macOS ships v3 of bash by default and many users won't bother updating it.

Reply:
Actually it's already checked in the caller, that's why the function is named with this `AA` suffix and is called only when associative array is supported.

Reply:
Oh perfect 👍